assets/aml-benchmark/components/src/dataset_preprocessor/dataset_preprocessor.py

```diff
@@ -0,0 +1,135 @@
+            dict value is presented using jinja template logic which will be used to extract the \
+            respective value from the dataset.
+        :param user_preprocessor: Path to the custom preprocessor python script provided by user.
+        :param encoder_config: JSON serialized dictionary to perform mapping. Must contain key-value pair \
+            "column_name": "<actual_column_name>" whose value needs mapping, followed by key-value pairs containing \
+            idtolabel or labeltoid mappers. Example format: \
+            {"column_name":"label", "0":"NEUTRAL", "1":"ENTAILMENT", "2":"CONTRADICTION"}. This is not applicable to \
+            custom scripts.
+        :param output_dataset: Path to the jsonl file where the processed data will be saved.
+        """
+        self.input_dataset = input_dataset
+        self.template = template
+        self.user_preprocessor = user_preprocessor
+        self.encoder_config = encoder_config
+        self.output_dataset = output_dataset
+
+    def __post_init__(self) -> None:
+        """Post init call."""
+        self._validate()
+
+    def validate(self) -> None:
+        """Validate the parameters."""
+        if self.input_dataset is None:
+            mssg = (
+                "Path to load the dataset is not provided."
+            )
+            raise BenchmarkValidationException._with_error(
+                AzureMLError.create(BenchmarkValidationError, error_details=mssg)
+            )
+        if self.template is None and self.user_preprocessor is None:
+            mssg = (
+                "Please provide the input to apply preprocessing logic either via template input or \
+                script_path."
+            )
+            raise BenchmarkValidationException._with_error(
+                AzureMLError.create(BenchmarkValidationError, error_details=mssg)
+            )
+        if self.user_preprocessor and not self.user_preprocessor.endswith('.py'):
+            mssg = (
+                "Please provide python script containing your custom preprocessor logic."
+            )
+            raise BenchmarkValidationException._with_error(
+                AzureMLError.create(BenchmarkValidationError, error_details=mssg)
+            )
+
+    def add_json_filter(self, template) -> str:
+        """Add tojson filter in the template."""
+        patt = r'({{.*?}})'
+        pat = re.compile(patt)
+        matches = re.findall(pat, template)
+        for m in range(len(matches)):
+            logger.info(f"{template}, {matches[m]}")
+            s1 = '{{'+matches[m].lstrip('{{').rstrip('}}')+'|tojson}}'
+            template = template.replace(matches[m], s1)
+        logger.info(f"Final template:{template}")
+        return template
+
+    def prep_using_template(self) -> None:
+        """Preprocessor run using template."""
+        data = read_jsonl_files([self.input_dataset])

```

Comment:
Instead of using `read_jsonl_files` files, we need to `resolve_io_path` like in dataset sampler. @iamrk04 , can you share the proper approach to load the data? Also, please ensure we are doing the same in postprocessor.

Reply:
This is created with assumption that I only support the processing of only one jsonl file, and we are going to support that only based on the design. This is the reason resolve io path not called as I have clearly mentioned it in the parameter's description also.

Reply:
One example to properly read the file(s) can be found [here](https://github.com/Azure/azureml-assets/blob/53e0ca677ca79133c41dbec078a21f3582fcb201/assets/aml-benchmark/components/src/compute_performance_metrics/main.py#L231C3-L231C3).

The idea is to resolve the path first to get a list of paths to file, and then use the `read_jsonl_files`. In this case it should be:
```
  input_file_paths = resolve_io_path(self.input_dataset)
  data = read_jsonl_files(input_file_paths)
```

@Nivi09. please update this, we never know when we will need to allow passing `mltable` or even `uri_folder`, this way we are already prepared for the future.

Reply:
Done. Thanks.